native/src/components/CalendarRangeModal.tsx

```diff
@@ -0,0 +1,121 @@
+import { DateTime } from 'luxon'
+import React, { ReactElement, useEffect, useState } from 'react'
+import { useTranslation } from 'react-i18next'
+import { Modal } from 'react-native'
+import { Calendar } from 'react-native-calendars'
+import styled, { useTheme } from 'styled-components/native'
+
+import { getMarkedDates } from '../utils/calendarRangeUtils'
+import Caption from './Caption'
+import TextButton from './base/TextButton'
+
+const DatePickerWrapper = styled.View`
+  background-color: ${props => props.theme.colors.textDecorationColor};
+  border-radius: 20px;
+  position: absolute;
+  width: 90%;
+  top: 228px;
+  align-self: center;
+`
+const StyledView = styled.View`
+  gap: 8px;
+  justify-content: ${props => (props.theme.contentDirection === 'rtl' ? 'flex-start' : 'flex-end')};
+  flex-direction: ${props => (props.theme.contentDirection === 'rtl' ? 'row-reverse' : 'row')};
+  padding: 5px 10px;
+`
+const OPACITY_DISABLED = 0.5
+
+const StyledTextButton = styled(TextButton)`
+  background-color: transparent;
+  transform: scale(0.8);
+  opacity: ${props => (props.disabled ? OPACITY_DISABLED : 1)};
+`
+const StyledPressable = styled.Pressable`
+  flex: 1;
+`
+
+export type CalendarViewerProps = {
+  modalVisible: boolean
+  closeModal: () => void
+  startDate: DateTime | null
+  endDate: DateTime | null
+  setStartDate: (startDate: DateTime) => void
+  setEndDate: (endDate: DateTime) => void
+}
+
+const CalendarRangeModal = ({
+  modalVisible,
+  closeModal,
+  startDate,
+  endDate,
+  setStartDate,
+  setEndDate,
+}: CalendarViewerProps): ReactElement => {
+  const [tempStartDate, setTempStartDate] = useState<DateTime | null>(startDate)
+  const [tempEndDate, setTempEndDate] = useState<DateTime | null>(endDate)
+  const { t } = useTranslation('events')
+  const theme = useTheme()
+
+  useEffect(() => {
+    setTempStartDate(startDate)
```

Comment:
I think you should really take care of memoriziation here, as updating the state, will rerender the parent (Events.ts) and all of its children. Please check useMemo and useCallback hooks

Reply:
Hmm what would we use `useMemo` for? I think there is no value that would need wrapping. Also, `useMemo` is only helpful for resource-intensive calculations or e.g. objects (that are passed as props/contexts) as it will only prevent a recalculation of a value if the component already rerenders.

I'd have to test this but I think that also wrapping the functions in `useCallback` might not really help with anything as this only prevents rerendering of the children, not of the parent/siblings for the same reasons.

Reply:
I used useEffect just to listen for changes from outside of this component if you have a better way of doing this please let me know.
![explanation](https://github.com/user-attachments/assets/76a24355-c87c-49ea-88ef-b21e6c734333)

Reply:
Ah, okay, thank you :)